app/controllers/application_controller.rb

```diff
@@ -1,3 +1,21 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  before_action :authenticate_user!, :except => [:index]
```

Comment:
We need to decide whether we do this in every controller or just here @maciek1995